Source/path.cpp

```diff
@@ -434,63 +331,54 @@ void __fastcall path_next_node(PATHNODE *pPath)
 /* update all path costs using depth-first search starting at pPath */
 void __fastcall path_set_coords(PATHNODE *pPath)
 {
-	PATHNODE *PathOld; // edi
-	PATHNODE *PathAct; // esi
-	char next_g; // al
-	int i; // [esp+0h] [ebp-8h]
-	PATHNODE **child_ptr; // [esp+4h] [ebp-4h]
+	PATHNODE *PathOld;
+	PATHNODE *PathAct;
+	int i;
 
 	path_push_active_step(pPath);
-	while ( gdwCurPathStep )
-	{
+	while ( gdwCurPathStep ) {
 		PathOld = path_pop_active_step();
-		child_ptr = PathOld->Child;
-		for(i = 0; i < 8; i++)
-		{
-			PathAct = *child_ptr;
-			if ( !*child_ptr )
+		for(i = 0; i < 8; i++) {
+			PathAct = PathOld->Child[i];
+			if ( !PathAct )
 				break;
 
-			if ( PathOld->g + path_check_equal(PathOld, PathAct->x, PathAct->y) < PathAct->g )
-			{
-				if ( path_solid_pieces(PathOld, PathAct->x, PathAct->y) )
-				{
+			if ( PathOld->g + path_check_equal(PathOld, PathAct->x, PathAct->y) < PathAct->g ) {
+				if ( path_solid_pieces(PathOld, PathAct->x, PathAct->y) ) {
 					PathAct->Parent = PathOld;
-					next_g = PathOld->g + path_check_equal(PathOld, PathAct->x, PathAct->y);
-					PathAct->g = next_g;
-					PathAct->f = next_g + PathAct->h;
+					PathAct->g = PathOld->g + path_check_equal(PathOld, PathAct->x, PathAct->y);
+					PathAct->f = PathAct->g + PathAct->h;
 					path_push_active_step(PathAct);
 				}
 			}
-			++child_ptr;
 		}
 	}
 }
 
 /* push pPath onto the pnode_tblptr stack */
 void __fastcall path_push_active_step(PATHNODE *pPath)
 {
-	int stack_index; // eax
-
-	stack_index = gdwCurPathStep++;
+	int stack_index = gdwCurPathStep;
+	gdwCurPathStep++;
 	pnode_tblptr[stack_index] = pPath;
 }
 
 /* pop and return a node from the pnode_tblptr stack */
 PATHNODE *__cdecl path_pop_active_step()
 {
-	return pnode_tblptr[--gdwCurPathStep];
+	gdwCurPathStep--;
+	return pnode_tblptr[gdwCurPathStep];
 }
 
 /* zero one of the preallocated nodes and return a pointer to it, or NULL if
  * none are available */
 PATHNODE *__cdecl path_new_step()
 {
-	PATHNODE *new_node; // esi
-
 	if ( gdwCurNodes == 300 )
-		return 0;
-	new_node = &path_nodes[gdwCurNodes++];
-	memset(new_node, 0, 0x34u);
+		return NULL;
+
+	PATHNODE *new_node = &path_nodes[gdwCurNodes];

```

Comment:
No need to move the post-increment out of this line

Reply:
I'm not a fan of doing multiple things in one line.

Reply:
Someone had already cleaned this code and these three functions were already bin exact - there was no need to mess with the pre/postfix increments just because you didn't like how they looked. Plus now there's a variable declaration in the body of the function which would not have worked in the original code (compiled as plain C) so it's definitely not correct.

Reply:
Sorry If it seams that I'm just doing things out of personal preference, separating inline pre/postfix increments has been requested of me in previous PRs so I was just trying to follow the guidelines that I have been given.